src/Modules/OnPremisesDataGateway/Commands.OnPremisesDataGateway.Test/GetOnPremisesDataGatewayClusterTests.cs

```diff
@@ -0,0 +1,195 @@
+                TestUtilities.AssertNoCmdletErrors(ps);
+                Assert.IsNotNull(result);
+            }
+        }
+
+        [TestMethod]
+        [TestCategory("Interactive")]
+        [TestCategory("SkipWhenLiveUnitTesting")] // Ignore for Live Unit Testing
+        public void EndToEndGetOnPremisesDataGatewayClusterWithGatewayClusterId()
+        {
+            using (var ps = System.Management.Automation.PowerShell.Create())
+            {
+                // Arrange
+                ProfileTestUtilities.ConnectToPowerBI(ps);
+                ps.AddCommand(GetOnPremisesDataGatewayClusterInfo).AddParameter(nameof(GetOnPremisesDataGatewayCluster.GatewayClusterId), Guid.NewGuid());
+
+                // Act
+                var result = ps.Invoke();
+
+                // Assert
+                TestUtilities.AssertNoCmdletErrors(ps);
+                Assert.IsNotNull(result);
+            }
+        }
+
+        [TestMethod]
+        public void GetOnPremisesDataGatewayClusterReturnsExpectedResults()
+        {
+            // Arrange
+            var clusterId = Guid.NewGuid();
+            var expectedResponse = new GatewayCluster
+            {
+                Id = clusterId,
+                Name = "the cluster name",
+                Description = "the cluster description",
+                Status = "the status",
+                Region = "the region",
+                Permissions = new Permission[]
+                {
+                    new Permission
+                    {
+                        Id = Guid.NewGuid().ToString(),
+                        PrincipalType = "User",
+                        TenantId = Guid.NewGuid().ToString(),
+                        Role = "Admin",
+                        AllowedDataSources = new string[]
+                        {
+                            Guid.NewGuid().ToString()
+                        },
+                        ClusterId = clusterId.ToString()
+                    }
+                },
+                DataSourceIds = new Guid[]
+                {
+                    Guid.NewGuid()
+                },
+                Type = "the type",
+                MemberGateways = new MemberGateway[]
+                {
+                    new MemberGateway
+                    {
+                        Id = clusterId,
+                        Name = "the member name",
+                        Description = "the member description",
+                        Status = "the member status",
+                        Region = "the member region",
+                        Type = "the member type",
+                        Version = "the version",
+                        Annotation = "the annotation",
+                        ClusterId = clusterId
+                    }
+                }
+            };
+
+            var client = new Mock<IPowerBIApiClient>();
+            client.Setup(x => x.Gateways
+                .GetGatewayClusters(It.IsAny<Guid>(), true))
```

Comment:
Whenever possible, you should avoid using Any. You should be able to use clusterId here. You might want to make a pass through your other tests to correct this.

Reply:
Agreed. I realized I made a silly mistake and forgot to use the clusterId I defined up top in the actual request.